pkg/logql/syntax/serialize.go

```diff
@@ -304,6 +308,46 @@ func (v *JSONSerializer) VisitPipeline(e *PipelineExpr) {
 	v.Flush()
 }
 
+func (v *JSONSerializer) VisitVariants(e *MultiVariantExpr) {
+	v.WriteObjectStart()
+
+	v.WriteObjectField(Variants)
+	v.WriteObjectStart()
+
+	v.WriteObjectField(LogSelector)
+
+	// Serialize log range as string.
+	v.WriteObjectStart()
+	v.WriteObjectField(IntervalNanos)
+
+	logRange := e.LogRange()
+	v.WriteInt64(int64(logRange.Interval))
+	v.WriteMore()
+	v.WriteObjectField(OffsetNanos)
+	v.WriteInt64(int64(logRange.Offset))
+
+	// Serialize log selector pipeline as string.
+	v.WriteMore()
+	v.WriteObjectField(LogSelector)
+	encodeLogSelector(v.Stream, logRange.Left)
```

Comment:
No need to duplicate serialisation logic of the `LogRange`

```suggestion
    e.LogRange().Accept(v)
```

or

```suggestion
	v.VisitLogRange(e.LogRange())
```

Reply:
good catch, I'm going apply all of the suggestions in a single commit with the other fixes.